### PR TITLE
Fix unique backup label generation for diff and incr backup types

### PR DESCRIPTION
**Pull request by andr-sokolov**
If there were at least 2 full backups and then the last one was expired, then it was impossible to create either a differential or an incremental backup. The backupLabelCreate function identified this situation as clock skew, because the new backup label was compared with label of the expired full backup. If the new backup is differential or incremental, then its label should be compared with labels of differential or incremental backups related to the same full backup.
Replace numbers with macro.

### CODE REVIEW

**Comment by Stolb27** [comment]
@andr-sokolov, what about test case?

File: test/src/module/command/backupTest.c [hunk 1]
```diff
@@ -1428,6 +1428,7 @@ testRun(void)
 
         time_t timestamp = 1575401652;
         String *backupLabel = backupLabelFormat(backupTypeFull, NULL, timestamp);
+        const String *const olderBackupLabel = backupLabelFormat(backupTypeFull, NULL, timestamp - 3);
```
Comment (InnerLife0) [inline test/src/module/command/backupTest.c:1431]: Can you clarify why we've changed `- 2` to `- 3`?

File: src/command/backup/common.h [hunk 2]
```diff
@@ -14,6 +14,8 @@ Common Functions and Definitions for Backup and Expire Commands
 Backup constants
 ***********************************************************************************************************************************/
 #define BACKUP_PATH_HISTORY                                         "backup.history"
+#define DATE_TIME_REGEX                                             "[0-9]{8}\\-[0-9]{6}"
```
Comment (InnerLife0) [inline src/command/backup/common.h:19]: Maybe add a little comment with example of expected datetime and it's format (%Y%m%d-%H%M%S)?

File: test/src/module/command/backupTest.c (hunk 1 above)
Comment (andr-sokolov) [reply test/src/module/command/backupTest.c:1431]: -2 is used in line 1485, -1 - in 1504. Diff and incr backups should be after full backup. So -3 is used for full backup

File: src/command/backup/common.h (hunk 2 above)
Comment (andr-sokolov) [reply src/command/backup/common.h:19]: done

**Review by InnerLife0 — APPROVED** [review]

**Pull request merged**